### PR TITLE
[Bug] check base & target is undefined/null and fixed types

### PR DESCRIPTION
**Pull request by neighborhood999**
## Screenshots

截圖 2022-08-30 09 24 09

截圖 2022-08-30 09 24 34

## Description

- as title

Signed-off-by: Jie Peng <im@jiepeng.me>

**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

### CODE REVIEW

File: static_report/src/types/index.ts [hunk 1]
```diff
@@ -4,11 +4,24 @@ import {
...
+export interface SaferTableSchema extends Omit<TableSchema, 'columns'> {
+  columns: { [k: string]: ColumnSchema | undefined };
+}
 
 export interface ComparisonReportSchema {
-  base: SingleReportSchema;
-  input: SingleReportSchema; //old code: future key will be `target`
+  base: SaferSRSchema;
+  input: SaferSRSchema;
+  // base: SingleReportSchema;
```
Comment (jonycfu) [inline static_report/src/types/index.ts:23]: remove commented code

File: static_report/src/components/ComparisonReport/CRTableList/index.tsx [hunk 2]
```diff
@@ -43,7 +55,7 @@ export function CRTableList({
 }) {
   const { base, input: target } = data;
   const tables = transformAsNestedBaseTargetRecord<
-    SingleReportSchema['tables'],
+    SaferSRSchema['tables'],
     TableSchema
```
Comment (jonycfu) [inline static_report/src/components/ComparisonReport/CRTableList/index.tsx:59]: `SaferTableSchema`?

File: static_report/src/components/ComparisonReport/CRTableList/CRTableListColumnList.tsx [hunk 3]
```diff
@@ -3,45 +3,57 @@ import {
...
-  targetTableDatum: TableSchema;
+  baseTableDatum?: SaferTableSchema;
+  targetTableDatum?: SaferTableSchema;
 }) {
   const transformedData = transformAsNestedBaseTargetRecord<
-    TableSchema['columns'],
-    ColumnSchema
-  >(baseTableDatum.columns, targetTableDatum.columns);
+    SaferTableSchema['columns'],
+    ColumnSchema | undefined
```
Comment (jonycfu) [inline static_report/src/components/ComparisonReport/CRTableList/CRTableListColumnList.tsx:19]: the `undefined` here isn't needed as we specify in the method generics

File: static_report/src/components/ComparisonReport/CRTableList/CRTableListAssertions.tsx [hunk 4]
```diff
@@ -168,9 +181,12 @@ export function CRTargetTableAssertionsDifference({
...
       <Text as="span" color={isGreaterThanZero ? 'black' : 'inherit'}>
-        {difference === 0 && targetAssertions}
-        {difference < 0 && targetAssertions}
-        {difference > 0 && targetAssertions}
+        {difference === 0 &&
+          formatColumnValueWith(targetAssertions, formatNumber)}
+        {difference < 0 &&
+          formatColumnValueWith(targetAssertions, formatNumber)}
+        {difference > 0 &&
+          formatColumnValueWith(targetAssertions, formatNumber)}
```
Comment (jonycfu) [inline static_report/src/components/ComparisonReport/CRTableList/CRTableListAssertions.tsx:189]: why are the render conditions necessary if the arguments are the same for `formatColumnValueWidth`? 🤔

File: static_report/src/components/ComparisonReport/CRTabProfilingDetails.tsx [hunk 5]
```diff
@@ -18,8 +18,8 @@ export function CRTabProfilingDetails({
   zReport(ZTableSchema.safeParse(targetTable));
 
   const transformedData = transformAsNestedBaseTargetRecord<
-    TableSchema['columns'],
-    ColumnSchema
+    SaferTableSchema['columns'],
+    ColumnSchema | undefined
```
Comment (jonycfu) [inline static_report/src/components/ComparisonReport/CRTabProfilingDetails.tsx:22]: `undefined` no longer needed here

**Review by jonycfu — COMMENTED** [review]
minor stuff

File: static_report/src/components/ComparisonReport/CRTableList/index.tsx (hunk 2 above)
Comment (neighborhood999) [reply static_report/src/components/ComparisonReport/CRTableList/index.tsx:59]: Here is a table, I think using `SafterSRSchema` is right?

File: static_report/src/components/ComparisonReport/CRTableList/CRTableListAssertions.tsx (hunk 4 above)
Comment (neighborhood999) [reply static_report/src/components/ComparisonReport/CRTableList/CRTableListAssertions.tsx:189]: My fault, when i developed here and i list all conditions and I forgot to delete xd.

File: static_report/src/components/ComparisonReport/CRTableList/index.tsx (hunk 2 above)
Comment (jonycfu) [reply static_report/src/components/ComparisonReport/CRTableList/index.tsx:59]: Yup. I'm talking about the one after :) 
Line 59 `TableSchema` -> `SaferTableSchema`

Comment (neighborhood999) [reply static_report/src/components/ComparisonReport/CRTableList/index.tsx:59]: LOL, I saw the highlight and misunderstood, let me fix it.

**Review by jonycfu — APPROVED** [review]
final comment (minor) ☝️

**Pull request merged**